### PR TITLE
docs: add running and debugging pages

### PR DESCRIPTION
**Pull request by agnes512**
Implements https://github.com/strongloop/loopback-next/issues/4845

Include **Most of** the bug strings

Screen Shot 2020-05-05 at 1 49 58 PM

Screen Shot 2020-05-05 at 1 49 33 PM

Screen Shot 2020-05-05 at 4 16 01 PM

Screen Shot 2020-05-05 at 4 16 19 PM

Screen Shot 2020-05-05 at 4 16 31 PM

## Checklist

👉 [Read and sign the CLA (Contributor License Agreement)](https://cla.strongloop.com/agreements/strongloop/loopback-next) 👈

- [ ] `npm test` passes on your machine
- [ ] New tests added or existing tests modified to cover all changes
- [ ] Code conforms with the [style guide](http://loopback.io/doc/en/contrib/style-guide.html)
- [ ] API Documentation in code was updated
- [ ] Documentation in [/docs/site](../tree/master/docs/site) was updated
- [ ] Affected artifact templates in `packages/cli` were updated
- [ ] Affected example projects in `examples/*` were updated

👉 [Check out how to submit a PR](https://loopback.io/doc/en/lb4/submitting_a_pr.html) 👈

### CODE REVIEW

File: docs/site/Setting-debug-strings.md [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+
+For example:
+
+`loopback:cli:model-generator`
+
+The debug string `model-generator` is specified in the source file [`generators/model/index.js`](https://github.com/strongloop/loopback-next/blob/master/packages/cli/generators/model/index.js) of the `@loopback/cli` module.
+
+## Debug strings reference
+
+<table>
```
Comment (raymondfeng) [inline docs/site/Setting-debug-strings.md:81]: Did you create this table by hand? Can we automate this?

Comment (agnes512) [reply docs/site/Setting-debug-strings.md:81]: I don't know if we have any script to do it, so I checked modules one by one :flushed: And I didn't include all of the debug strings in here, or the list would be extremely long... I only include those important (from my understanding) ones. Any suggestions?

Comment (raymondfeng) [reply docs/site/Setting-debug-strings.md:81]: A shell script will do. Or monkey patch the debug function :-)

File: docs/site/Running-debugging-apps.md [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+---
+
+In general, when you are developing an application, use the `node` command to run it.
+This enables you to see stack traces and console output immediately.
+
+For example:
+
+```
+$ cd myapp
+$ node .
```
Comment (dhmlau) [inline docs/site/Running-debugging-apps.md:17]: not `npm start`?

File: docs/site/Setting-debug-strings.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+    </tr>
+    <tr>
+      <td></td>
+      <td>src/booters/model-api.booter.ts</td>
+      <td>loopback:boot:model-api</td>
+    </tr>
+    <tr>
+      <td></td>
+      <td>src/booters/service.booter.ts</td>
+      <td>loopback:boot:service-booterr</td>
```
Comment (dhmlau) [inline docs/site/Setting-debug-strings.md:119]: typo:
```suggestion
      <td>loopback:boot:service-booter</td>
```

File: docs/site/Setting-debug-strings.md (hunk 1 above)
Comment (emonddr) [reply docs/site/Setting-debug-strings.md:81]: @agnes512 ,

Here's a script I just made:

```sh
#!/bin/bash 
clear
FILE_EXTENSION=$1
SEARCH_TEXT=$2
echo ""
echo " Files with extension : '${FILE_EXTENSION}'  "
echo " that contains text   : '${SEARCH_TEXT}' : "
echo "----------------------------------------------"
echo ""
find . -name "${FILE_EXTENSION}" -exec grep -h "${SEARCH_TEXT}" {} +
echo ""
echo ""




# include file name and found search text
# find . -name "*.ts" -exec grep "debugFactory('loopback" {} +

# only show found search text
# find . -name "*.ts" -exec grep -h "debugFactory('loopback" {} +
```

Call

```sh
find_files_with_text.sh "*.ts" "'loopback:"
```

If you want to see :  filename searchText in the results, remove the `-h` in the script.
Right now it outputs ONLY the found search text :   searchText

Based on my search string above, I found:
```sh
 Files with extension : '*.ts'  
 that contains text   : ''loopback:' : 
----------------------------------------------

const debug = require('debug')('loopback:benchmark');
const debug = debugFactory('loopback:cron');
const debug = debugFactory('loopback:multi-tenancy:strategy:query');
const debug = debugFactory('loopback:multi-tenancy:strategy:header');
const debug = debugFactory('loopback:multi-tenancy:strategy:host');
const debug = debugFactory('loopback:multi-tenancy:strategy:jwt');
const debug = debugFactory('loopback:multi-tenancy:action');
const debug = require('debug')('loopback:example:acl');
const debug = debugFactory('loopback:boot:lb3app');
const debug = debugFactory('loopback:repository:belongs-to-accessor');
const debug = debugFactory('loopback:repository:belongs-to-helpers');
const debug = debugFactory('loopback:repository:has-one-helpers');
const debug = debugFactory('loopback:repository:has-one-repository-factory');
const debug = debugFactory('loopback:repository:has-many-repository-factory');
const debug = debugFactory('loopback:repository:has-many-inclusion-resolver');
const debug = debugFactory('loopback:repository:has-many-helpers');
const debug = debugFactory('loopback:repository:relation-helpers');
export const RELATIONS_KEY = 'loopback:relations';
const debug = debugFactory('loopback:repository:mixin');
>('loopback:model');
>('loopback:model-properties');
>('loopback:model-and-properties');
const debug = debugFactory('loopback:context:binding');
const debug = debugFactory('loopback:context:invocation');
const debug = debugFactory('loopback:context:interceptor');
const debug = debugModule('loopback:context:resolver');
const debug = debugFactory('loopback:context:binding-inspector');
   * const debug = debugFactory('loopback:context:application');
   *   return 'loopback:context:application';
    if (this.constructor === Context) return 'loopback:context';
const debug = debugFactory('loopback:context:subscription');
      expect(myCtx.debugFn.namespace).to.eql('loopback:context:mycontext');
const debug = debugModule('loopback:context:test');
const debug = debugFactory('loopback:context:interceptor-chain');
const debugSession = debugModule('loopback:context:resolver:session');
const debug = debugFactory('loopback:context:view');
const debug = debugFactory('loopback:core:lifecycle');
const debug = debugFactory('loopback:core:application');
const debugShutdown = debugFactory('loopback:core:application:shutdown');
const debugWarning = debugFactory('loopback:core:application:warning');
const debug = debugFactory('loopback:boot:booter-utils');
const debug = debugFactory('loopback:boot:base-artifact-booter');
const debug = debugFactory('loopback:boot:lifecycle-observer-booter');
const debug = debugFactory('loopback:boot:interceptor-booter');
const debug = debugFactory('loopback:boot:service-booter');
const debug = debugModule('loopback:boot:booter:application-metadata');
const debug = debugFactory('loopback:boot:model-api');
const debug = debugModule('loopback:boot:bootstrapper');
const debug = debugFactory('loopback:http-caching-proxy');
const debug = debugFactory('loopback:tsdocs');
const debug = debugFactory('loopback:tsdocs');
>('loopback:json-schema');
const debug = debugFactory('loopback:repository-json-schema:build-schema');
  'loopback:openapi3:metadata:controller-spec:deprecated',
const debug = require('debug')('loopback:openapi3:metadata:requestbody');
const debug = debugModule('loopback:openapi:spec-enhancer');
const debug = debugModule('loopback:openapi:spec-enhancer');
const debug = debugModule('loopback:openapi:spec-enhancer');
const debug = require('debug')('loopback:openapi3:metadata:controller-spec');
const debug = debugFactory('loopback:repository-tests');
const debug = debugFactory('loopback:authorization:interceptor');
const debug = debugModule('loopback:metadata:decorator');
const debug = debugFactory('loopback:boot:crud-rest');
const debug = debugFactory('loopback:openapi:spec-enhancer:consolidate');
const debug = debugFactory('loopback:openapi:spec-enhancer:info');
const debug = debugModule('loopback:rest:body-parser');
const debug = debugModule('loopback:rest:body-parser');
const debug = debugFactory('loopback:rest:parser');
const debug = debugFactory('loopback:openapi:spec-enhancer');
const debug = require('debug')('loopback:rest:sequence');
const debug = debugFactory('loopback:rest:server');
const debug = debugModule('loopback:rest:coercion');
const debug = debugModule('loopback:rest:coercion');
const debug = debugModule('loopback:rest:ajv');
const debug = debugModule('loopback:rest:validation');
const debug = require('debug')('loopback:rest:router:trie');
const debug = debugFactory('loopback:rest:routing-table');
const debug = require('debug')('loopback:rest:router:regexp');
const debug = debugFactory('loopback:rest:controller-route');

```

but feel free to adjust the search string

Comment (agnes512) [reply docs/site/Setting-debug-strings.md:81]: Thanks!! I also found that some packages have different ways to define the string e.g [`@loopback:cli`](https://github.com/strongloop/loopback-next/blob/master/packages/cli/lib/debug.js) 😂 which I have no idea how to write a script for

File: docs/site/Running-debugging-apps.md (hunk 2 above)
Comment (agnes512) [reply docs/site/Running-debugging-apps.md:17]: Do we want to add API Connect here?

Comment (dhmlau) [reply docs/site/Running-debugging-apps.md]: i don't think so.

File: docs/site/Running-debugging-apps.md [hunk 4]
```diff
@@ -0,0 +1,22 @@
+---
+title: 'Running and debugging apps'
+lang: en
+keywords: LoopBack 4.0, LoopBack 4, Debug
+sidebar: lb4_sidebar
+permalink: /doc/en/lb4/Running-and-debugging-apps.html
+summary:
+---
+
+In general, when you are developing an application, use the `node` command to
```
Comment (dhmlau) [inline docs/site/Running-debugging-apps.md:10]: we might want to be consistent to use `npm start` throughout the docs.

File: docs/site/Setting-debug-strings.md [hunk 5]
```diff
@@ -0,0 +1,329 @@
...
+
+## Using debug strings
+
+LoopBack has a number of built-in debug strings to help with debugging.Specify a
+string on the command-line via an environment variable as follows:
+
+**MacOS and Linux**
+
+```shell
+$ DEBUG=<pattern>[,<pattern>...] node .
```
Comment (dhmlau) [inline docs/site/Setting-debug-strings.md:23]: might want to change it to use `npm start` instead.

File: docs/site/Setting-debug-strings.md [hunk 6]
```diff
@@ -0,0 +1,329 @@
...
+---
+
+You can specify debug strings when you run an application, as explained below,
+to display specific log output to the console. You can also redirect the output
+to a file, if desired. These techniques are often helpful in debugging
+applications.
+
+## Using debug strings
+
+LoopBack has a number of built-in debug strings to help with debugging.Specify a
```
Comment (jannyHou) [inline docs/site/Setting-debug-strings.md:17]:
```suggestion
LoopBack has a number of built-in debug strings to help with debugging. Specify a
```

Comment (agnes512) [reply docs/site/Setting-debug-strings.md:17]: oops, will fix it in https://github.com/strongloop/loopback-next/issues/4846

**Pull request merged**